sandbox/cgroup/memory.go

```diff
@@ -38,9 +49,30 @@ var cgroupsFilePath = "/proc/cgroups"
 // 0 => false => disabled
 // 1 => true => enabled
 func CheckMemoryCgroup() error {
-	cgroupsFile, err := os.Open(cgroupsFilePath)
+	var supp bool
+	var err error
+	if IsUnified() {
+		supp, err = checkV2CgroupMemoryController()
+	} else {
+		supp, err = checkV1CgroupMemoryController()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if supp {
+		return nil
+	}
+
+	// no errors so far but the only path here is the cgroups file without the memory line
+	return fmt.Errorf("cgroup memory controller is disabled on this system")
+}
+
+func checkV1CgroupMemoryController() (bool, error) {
+	cgroupsFile, err := os.Open(filepath.Join(rootPath, cgroupV1ControllersPath))
 	if err != nil {
-		return fmt.Errorf("cannot open cgroups file: %v", err)
+		return false, fmt.Errorf("cannot open cgroups file: %v", err)
```

Comment:
We can wrap the error:
```suggestion
		return false, fmt.Errorf("cannot open cgroups file: %w", err)
```